src/components/Stocks.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { Link, useNavigate } from 'react-router-dom';
+import ShowButton from './ShowButton';
+import { categories } from './data';
+import Loading from './Loading';
+import '../styles/Stocks.css';
+
+function Stocks() {
+  const stocks = useSelector((state) => state.stocks);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [itemsToShow, setItemsToShow] = useState(4);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (itemsToShow === 4 || itemsToShow === stocks.length) {
+      setIsLoading(false);
+    }
+  }, [isExpanded]);
+
+  const show = () => {
+    if (itemsToShow === 4) {
+      setItemsToShow(stocks.length);
+      setIsExpanded(true);
+      setIsLoading(true);
+    } else {
+      setItemsToShow(4);
+      setIsExpanded(false);
+    }
+  };
+
+  return (
+    <section className="stocks-page">
+      <div className="top-bar">
+        <button
+          type="button"
+          className="go-back"
+          onClick={() => navigate('/')}
+        >
+          <i className="chevron left icon" />
+        </button>
```

Comment:
- [ ] As it is requirement for this project, which also improves the user experience of your app, kindly ensure  there's a link(`"Back" button`) in the `Detail page` that you can click on to go to the `Home page`. This button is absent in your app, please kindly fix that. See screenshot of the requirements below;

![backbutton](https://user-images.githubusercontent.com/65852621/152334021-f32e00b9-6377-448a-87f3-e1aa6b61a3e4.png)